From 025866ef6870a6be1381022461a9db374c0cb040 Mon Sep 17 00:00:00 2001
From: xiaoming wang <xiaoming.wang@intel.com>
Date: Sun, 19 Aug 2012 17:20:37 -0400
Subject: [PORT FROM MAIN] User Stack: User space stack is not integrated

BZ: 51255

The upgrade of gcc compiler cause register ebp used as normal register
because compile option -fomit-frame-pointer used as default option
in compiler. So should add compile option  -fno-omit-frame-pointer
to dump stack integrated in user space.

Source: Internal

Change-Id: I35f6dbbf618a278e434860779fc7bb5bd6efde19
Orig-Change-Id: Icc1d6eef9541f2ed94ca1b335244e1de5ffb2231
Category: device enablement
Domain: AOSP-bionic
Origin: internal
Upstream-Candidate: yes
Signed-off-by: xiaoming wang <xiaoming.wang@intel.com>
Signed-off-by: xindong ma<xindong.ma@intel.com>
Signed-off-by: Christophe Guerard <christophe.guerard@intel.com>
---
 core/combo/TARGET_linux-x86.mk |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/core/combo/TARGET_linux-x86.mk b/core/combo/TARGET_linux-x86.mk
index ab02902..f058b4c 100644
--- a/core/combo/TARGET_linux-x86.mk
+++ b/core/combo/TARGET_linux-x86.mk
@@ -97,6 +97,10 @@ KERNEL_HEADERS := $(KERNEL_HEADERS_COMMON) $(KERNEL_HEADERS_ARCH)
 
 android_config_h := $(call select-android-config-h,target_linux-x86)
 
+ifneq ($(TARGET_BUILD_VARIANT),user)
+TARGET_GLOBAL_CFLAGS += -fno-omit-frame-pointer
+endif
+
 TARGET_GLOBAL_CFLAGS += \
 			-O2 \
 			-Ulinux \
-- 
1.7.4.1

